packages/account/src/Containers/Account/page-overlay-wrapper.tsx

```diff
@@ -48,8 +48,10 @@ const PageOverlayWrapper = observer(({ routes, subroutes }: PageOverlayWrapperPr
             passkeysMenuCloseActionEventTrack();
         }
 
-        is_next_wallet_enabled ? history.push(shared_routes.wallets) : history.push(shared_routes.traders_hub);
-    }, [history, is_next_wallet_enabled]);
+        has_wallet || is_next_wallet_enabled
+            ? history.push(shared_routes.wallets)
+            : history.push(shared_routes.traders_hub);
+    }, [history, has_wallet, is_next_wallet_enabled]);
```

Comment:
When `has_wallet === true`, this is the case and the flag is not being checked because the first part of "or" expression is `true`
When not migrated, logic is the same as it was